### DISCUSSION TITLE
Question about REPL

### DISCUSSION THREAD

**Discussion by ekickx** [Olical/conjure#593] [Q&A]
Hey I'm new to clojure and still don't understand about repl, nrepl, etc. In the wiki, [it](https://github.com/Olical/conjure/wiki/Quick-start:-Clojure-(babashka)#optional-start-with-an-nrepl-server-from-the-command-line) said that conjure will automatically start babashka's nrepl server. But I can't evaluate unless I manually create and connect babashka's nrepl server, neovim nightly and conjure v4.52.2

https://github.com/user-attachments/assets/d4b8de5e-fab9-4642-8639-2888277da07e

I also follow [this wiki](https://github.com/Olical/conjure/wiki/Quick-start:-Clojure). What's the difference between babashka's nrepl and `nrepl/nrepl` for evaluating?

https://github.com/user-attachments/assets/70844c59-39e0-45d2-96c7-19be47293f57

**Comment by russtoku** [comment]
The nrepl/nrepl should be the same as the babashka nrepl.

Your first video shows that the `bb nrepl-server` is started when you edit main.clj but doesn't respond when you evaluate forms. This would suggest that Conjure can't read the `.nrepl-port` file to know what port to connect to. In the directory where you ran `e main.clj`, there should be a `.nrepl-port` file.

If you don't see that file, then likely Windows can't create it. You can change that name to something else by setting the Neovim variable, `g:conjure#client#clojure#nrepl#connection#auto_repl#port_file`. For Linux and macOS, the period at the beginning of the filename is used to hide the file from casual browsing of a directory. In this case, it's not really necessary so you can just leave off the period or use any name you like.

**Reply by Olical** [reply]
It may well be that on Windows Conjure starts babashka, writes the port file (which Conjure does itself, babashka doesn't do this for us) but then can't find the file due to Windows filesystem differences.

@ekickx, when you're running Conjure and it says it started babashka, can you see a `.nrepl-port` file in your directory? And if so, what does it contain?

Could you also try doing this but with `:let g:conjure#debug = v:true` set before you open a Clojure file? If you can do that and paste the log output here that should help a lot in diagnosing what's going on.

With Windows, 99% of the time it's filesystem / path related issues sadly.
